### PR TITLE
add video recognition

### PR DESCRIPTION
**Pull request by haiasd**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

# Motivation
Please describe the motivation of this PR and the goal you want to achieve through this PR.

# Modification
Please briefly describe what modification is made in this PR.

# Backward-Compatibility-Breaking (Optional)
Does the modification introduce changes that break the backward-compatibility? If so, please describe how it breaks the compatibility in this PR.

# Use cases (Optional)
If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

# Pull-Request Todo List
1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: easycv/datasets/video_recognition/pipelines/loading.py [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+        self.num_clips = num_clips
+        self.temporal_jitter = temporal_jitter
+        self.twice_sample = twice_sample
+        self.out_of_bound_opt = out_of_bound_opt
+        self.test_mode = test_mode
+        self.frame_uniform = frame_uniform
+        assert self.out_of_bound_opt in ['loop', 'repeat_last']
+
+        if start_index is not None:
+            warnings.warn('No longer support "start_index" in "SampleFrames", '
```
Comment (Cathy0908) [inline easycv/datasets/video_recognition/pipelines/loading.py:142]: 这个warning删除，并删除 start_index参数

File: easycv/datasets/video_recognition/pipelines/loading.py [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+                    f'frame_interval={self.frame_interval}, '
+                    f'num_clips={self.num_clips}, '
+                    f'temporal_jitter={self.temporal_jitter}, '
+                    f'twice_sample={self.twice_sample}, '
+                    f'out_of_bound_opt={self.out_of_bound_opt}, '
+                    f'test_mode={self.test_mode})')
+        return repr_str
+
+
+def to_tensor(data):
```
Comment (Cathy0908) [inline easycv/datasets/video_recognition/pipelines/loading.py:288]: 接口已存在：easycv/datasets/shared/pipelines/format.py

File: easycv/datasets/video_recognition/raw.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        try:
+            data_dict = self.data_source[idx]
+            data_dict = self.pipeline(data_dict)
+        except:
+            logging.error(
+                'When parsing line {}, error happened with msg: {}'.format(
+                    idx, traceback.format_exc()))
+            data_dict = None
+        if data_dict is None:
+            return self.__getitem__(np.random.randint(self.__len__()))
```
Comment (Cathy0908) [inline easycv/datasets/video_recognition/raw.py:38]: return self[np.random.randint(len(self))]

File: easycv/models/backbones/clip_bert.py [hunk 4]
```diff
@@ -0,0 +1,582 @@
+# flake8: noqa
```
Comment (Cathy0908) [inline easycv/models/backbones/clip_bert.py:1]: 加一下引用的lincense

File: easycv/models/backbones/swin_transformer3d.py [hunk 5]
```diff
@@ -0,0 +1,732 @@
+from functools import lru_cache, reduce
```
Comment (Cathy0908) [inline easycv/models/backbones/swin_transformer3d.py:1]: 加一下引用的license

File: easycv/models/backbones/x3d.py [hunk 6]
```diff
@@ -0,0 +1,333 @@
+import math
```
Comment (Cathy0908) [inline easycv/models/backbones/x3d.py:1]: 添加引用license

File: configs/video_recognition/clipbert/clipbert_multilabel.py [hunk 7]
```diff
@@ -0,0 +1,173 @@
+_base_ = '../../base.py'
```
[the remark this answers is not in the document]
Comment (haiasd) [reply configs/video_recognition/clipbert/clipbert_multilabel.py:1]: clipbert中的multi_class参数没有用到，默认是false，已改成fasle

File: easycv/core/evaluation/classification_eval.py [hunk 8]
```diff
@@ -147,4 +147,97 @@ def _evaluate_impl(self, predictions, gt_labels):
         return eval_res
 
 
+@EVALUATORS.register_module
+class MultiClsEvaluator(Evaluator):
```
[the remark this answers is not in the document]
Comment (haiasd) [reply easycv/core/evaluation/classification_eval.py:151]: done

File: easycv/core/evaluation/classification_eval.py [hunk 9]
```diff
@@ -147,4 +147,97 @@ def _evaluate_impl(self, predictions, gt_labels):
...
+        elif not (isinstance(pred, np.ndarray)
+                  and isinstance(target, np.ndarray)):
+            raise TypeError('pred and target should both be torch.Tensor or'
+                            'np.ndarray')
+
+        assert pred.shape == \
+            target.shape, 'pred and target should be in the same shape.'
+        num_classes = pred.shape[1]
+        ap = np.zeros(num_classes)
+        f = open('res.txt', 'w')
```
[the remark this answers is not in the document]
Comment (haiasd) [reply easycv/core/evaluation/classification_eval.py:199]: 已删除

**Review by Cathy0908 — APPROVED** [review]

**Pull request merged**